chain/network/src/routing_table_actor.rs

```diff
@@ -14,22 +16,322 @@ use crate::ibf::{Ibf, IbfBox};
+            // We already have a newer information about this edge. Discard this information.
+            false
+        } else {
+            self.edges_changed = true;
+            match edge.edge_type() {
+                EdgeType::Added => {
+                    self.raw_graph.add_edge(key.0.clone(), key.1.clone());
+                }
+                EdgeType::Removed => {
+                    self.raw_graph.remove_edge(&key.0, &key.1);
+                }
+            }
+            #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+            self.peer_ibf_set.add_edge(&edge.to_simple_edge());
+            self.edges_info.insert(key, edge);
+            true
+        }
+    }
+
+    /// Add several edges to the current view of the network.
+    /// These edges are assumed to be valid at this point.
+    /// Return list of edges added.
+    pub fn add_verified_edges_to_routing_table(&mut self, mut edges: Vec<Edge>) -> Vec<Edge> {
+        let total = edges.len();
+        edges.retain(|edge| {
+            let key = edge.get_pair();
+
+            self.touch(&key.0);
+            self.touch(&key.1);
+
+            self.add_edge(edge.clone())
+        });
+
+        // Update metrics after edge update
+        near_metrics::inc_counter_by(&metrics::EDGE_UPDATES, total as u64);
+        near_metrics::set_gauge(&metrics::EDGE_ACTIVE, self.raw_graph.total_active_edges as i64);
+
+        edges
+    }
+
+    /// If peer_id is not on memory check if it is on disk in bring it back on memory.
+    fn touch(&mut self, other_peer_id: &PeerId) {
+        if other_peer_id == self.peer_id()
+            || self.peer_last_time_reachable.contains_key(other_peer_id)
+        {
+            return;
+        }
+
+        let me = self.peer_id().clone();
+
+        if let Ok(nonce) = self.component_nonce_from_peer(other_peer_id.clone()) {
+            let mut update = self.store.store_update();
+
+            if let Ok(edges) = self.get_and_remove_component_edges(nonce, &mut update) {
+                for edge in edges {
+                    for &peer_id in vec![&edge.peer0, &edge.peer1].iter() {
+                        if peer_id == &me || self.peer_last_time_reachable.contains_key(peer_id) {
+                            continue;
+                        }
+
+                        if let Ok(cur_nonce) = self.component_nonce_from_peer(peer_id.clone()) {
+                            if cur_nonce == nonce {
+                                self.peer_last_time_reachable
+                                    .insert(peer_id.clone(), Instant::now() - SAVE_PEERS_MAX_TIME);
+                                update
+                                    .delete(ColPeerComponent, Vec::from(peer_id.clone()).as_ref());
+                            }
+                        }
+                    }
+                    self.add_edge(edge);
+                }
+            }
+
+            if let Err(e) = update.commit() {
```

Comment:
You are right, if there is a crash in the middle we will lose those edges. That's true for all active edges in memory, they are not stored anywhere.

I don't think this is an issue, but rather than odd behavior.